### ISSUE TITLE
Add language support for specifying `pub(...)` scope

### ISSUE THREAD

**Issue by Wodann** [mun-lang/mun#248]
By default, a type or function defined in a module is not accessible outside of its file and sub-modules. You can expand accessibility in three ways:

- `pub`: accessible within the package and externally (incl. marshalling to the host language)
- `pub(package)`: accessible within the package
- `pub(super)`: accessible within parent module and its sub-modules

For an example, see #249.

**Comment by baszalmstra** [comment]
Im working on this! 👍
